### ISSUE TITLE
Export winsize

### ISSUE THREAD

**Issue by platzhirsch** [grafov/m3u8#53]
When decoding a playlist with a window size bigger than 8 then an encode will result in a truncated playlist. This due to this line [here](https://github.com/grafov/m3u8/blob/master/reader.go#L155).

```go
media, err = NewMediaPlaylist(8, 1024) // TODO make it autoextendable
```

Would it be possible to update the `winsize` afterwards through `Count()` internally?

**Comment by bradleyfalzon** [comment]
Can you see #22 and let me know of your thoughts there?

**Comment by platzhirsch** [comment]
@bradleyfalzon Sorry, didn't realize there was already more or less the same discussion going on. Also my last edit on my issue didn't go through. Basically, I am agreeing with:

> Perhaps another PR could include options to overwrite the winsize as the decode function does set to 8, others (probably myself) might want to change this on the fly.

It should certainly be possible. Is there any reason why `winsize` doesn't get exported as part of the struct? My original issue is fixed by simply declaring the playlist as a VOD but the overlaying issue is the non-accessible `winsize` in my opinion.

I could draft a pull request.

**Comment by bradleyfalzon** [comment]
A PR to provide a getter and setter for winsize (`WinSize()` and `SetWinSize()`) would be great and reference  issue #22, I can do it in the next couple of days though. Just include a simple test to ensure it's set

If you do, can you call `p.buf.Reset()` too.

See this commit for something similar we've recently done for the version, in particular the test but ignore the master manifest stuff obviously: 321517c5b347f0af72fabc802419959b5975b347

Again, if you don't have time, I can do it, but might be a couple of days.

**Comment by bradleyfalzon** [comment]
#22 still will need to be resolved though, but this'll help you and others when they wish to set the win size (as I've needed to do, so this will help me).

**Comment by platzhirsch** [comment]
Also something which came to my mind is the media type `EVENT` for which you want a continuously growing window size because you must not remove URLs from it or it should probably be treated in the same way as `VOD`. Is that a separate issue?

**Comment by bradleyfalzon** [comment]
Yeah, best for a separate issue, but something like the following?

```
    p, _ := m3u8.NewMediaPlaylist(5, 5) // don't ignore error
    p.MediaType = m3u8.EVENT            // make this an event
    p.Closed = true                     // when finished make closed
    for i := 0; i < 5; i++ {
        _ = p.Append(fmt.Sprintf("test%d.ts", i), 6.0, "") // don't ignore error
    }
    fmt.Println(p.Encode())
```

**Comment by bradleyfalzon** [comment]
Hi @platzhirsch any progress on the PR, if you've moved on that's fine, I can manage it from here.

**Comment by platzhirsch** [comment]
Unfortunately not, no :person_frowning:

**Comment by bradleyfalzon** [comment]
No problem, do you still it? I don't want to add it unless someone needs it, although I'm sure someone else will want it some day for their use case.

**Comment by bradleyfalzon** [comment]
Hi @platzhirsch I've proposed some changes in #70 for the EVENT issue you mentioned and also adds the winsize getter and setter.

**Comment by platzhirsch** [comment]
This looks great @bradleyfalzon Thanks a lot for your effort.

**Comment by bradleyfalzon** [comment]
Thanks, I'll wait on a review from the maintainer and hopefully get it merged and close off a few issues. Thanks again for this input.
